packages/components/src/slot-fill2/context.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	createContext,
+	useMemo,
+	useCallback,
+	useState,
+	useContext,
+} from '@wordpress/element';
+
+export const SlotFillContext = createContext();
+
+function useSlotRegistry() {
+	const [ slots, setSlots ] = useState( {} );
```

Comment:
On master, we have a `getSlots` function whose reference will never change. By putting the `slots` object into state directly, this will automatically trigger an update/re-render on consumers (`Slot` and `Fill` components) when new slots are added, removed or updated (`fillProps`).

Reply:
This is neat. I like this proposal. It removes the need for the existence of the `occurrences` local variable.